src/reverse-proxy-handler.ts

```diff
@@ -71,6 +73,28 @@ function copyReverseOwnDescriptors(env: SecureEnvironment, shadowTarget: Reverse
     }
 }
 
+function callWithErrorBoundaryProtection(env: SecureEnvironment, fn: () => RawValue): RawValue {
+    let raw;
+    try {
+        raw = fn();
+    } catch (e) {
+        // This error occurred when the outer realm invokes a function from the sandbox.
+        if (e instanceof Error) {
+            throw e;
+        }
+        let rawError;
+        try {
+            rawError = new (env.getRawValue(e.constructor))(e.message);
```

Comment:
Would using `Reflect.construct` help here?

Reply:
is there any particular benefit on the new? it will require the creation of a brand new array for the arguments.